cli/src/semgrep/git.py

```diff
@@ -338,10 +341,10 @@ def baseline_context(self) -> Iterator[None]:
 
     def print_git_log(self) -> None:
         base_commit_sha = git_check_output(["git", "rev-parse", self._base_commit])
-        merge_base_sha = git_check_output(
-            ["git", "merge-base", self._base_commit, "HEAD"]
+        merge_base_sha = self._get_git_merge_base()
+        logger.info(
+            "  Will report findings introduced by these commits (may be incomplete for shallow checkouts):"
```

Comment:
is it possible to detect shallow checkout so that we print the additional warning only if we know it's a shallow checkout?

Reply:
I'm not sure. At this point I'm reluctant to complicate this further. Do you think it's critical?